### PR TITLE
feat(YNOProject): add Presence

### PR DESCRIPTION
**Pull request by luo3house**
## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

Screenshot 2024-11-10 at 12 27 25
Screenshot 2024-11-10 at 12 30 22
Screenshot 2024-11-10 at 12 31 28
Screenshot 2024-11-10 at 12 32 11
Screenshot 2024-11-10 at 12 32 57
Screenshot 2024-11-10 at 12 35 18
Screenshot 2024-11-10 at 12 45 39

</details>

### CODE REVIEW

**Review by darkvillager2 — APPROVED** [review]
LGTM

**Comment by luo3house** [comment]
Screenshot 2024-11-11 at 22 27 01

Screenshot 2024-11-11 at 22 19 58

<details>
<summary> See more </summary>



Screenshot 2024-11-11 at 22 23 21
Screenshot 2024-11-11 at 22 29 09
Screenshot 2024-11-11 at 22 26 13
Screenshot 2024-11-11 at 22 24 48
Screenshot 2024-11-11 at 22 23 50
Screenshot 2024-11-11 at 22 18 15


</details>

File: websites/Y/YNOProject/presence.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
+const PresenceAssets = {
+		Name: "YNOProject",
+		Logo: "https://imgur.com/2LD3PQV.png",
+	},
+	presence = new Presence({ clientId: "1304833580291063848" });
+// const strings = presence.getStrings({
+// 	play: "presence.playback.playing",
+// 	pause: "presence.playback.paused",
+// });
```
Comment (theusaf) [inline websites/Y/YNOProject/presence.ts:9]: Remove commented out code

File: websites/Y/YNOProject/presence.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+/**
+ * Read live favicon of character face in game.
+ * Size is scaled up from 16 to 40 to be sharper
+ * and encoded in Data URL.
+ *
+ * @returns Data URL or nothing at the portal
+ */
+async function fetchFavIcon(): Promise<string | void> {
+	const url = (
+		document.querySelector("#favicon") as HTMLLinkElement | undefined
```
Comment (theusaf) [inline websites/Y/YNOProject/presence.ts:45]:
```suggestion
		document.querySelector<HTMLLinkElement >("#favicon")
```

File: websites/Y/YNOProject/presence.ts [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+	}
+}
+
+/**
+ * The name of playing game, or nothing at the portal
+ * @example "Yume 2kki Online - YNOproject".match(it)?.[0] // Yume 2kki
+ */
+async function fetchGameName(): Promise<string | void> {
+	return (
+		document.querySelector("title") as HTMLElement | null
```
Comment (theusaf) [inline websites/Y/YNOProject/presence.ts:68]:
```suggestion
		document.querySelector("title")
```

File: websites/Y/YNOProject/presence.ts [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+	)?.textContent?.match(RegExp("^.+(?= Online -)"))?.[0];
+}
+
+/**
+ * Read current location within the game.
+ */
+async function fetchGameLocation(): Promise<string | void> {
+	return (document.querySelector("#locationText") as HTMLElement | null)
```
Comment (theusaf) [inline websites/Y/YNOProject/presence.ts:76]:
```suggestion
	return document.querySelector("#locationText")
```

File: websites/Y/YNOProject/presence.ts [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+	const presenceData: PresenceData = {
+		name: PresenceAssets.Name,
+		type: ActivityType.Playing,
+		startTimestamp: GameState.startedAt,
+		largeImageKey: await fetchFavIcon().then(url => url || PresenceAssets.Logo),
+		details: gameName || "Choosing a game...",
+		state: gameName ? gameLocation || "Disconnected" : void 0,
+		buttons: gameName
+			? [{ label: `Play ${gameName}`, url: location.href }]
```
Comment (theusaf) [inline websites/Y/YNOProject/presence.ts:24]:
```suggestion
			? [{ label: `Play ${gameName}`, url: document.location.href }]
```

**Review by theusaf — CHANGES_REQUESTED** [review]
Logic looks good, just a couple of stylistic things to fix to be consistent with other presence's quality and readability.

**Comment by luo3house** [comment]
Thanks for typo suggestions

**Comment by luo3house** [comment]
(Animated) badges for logged-in players
Screenshot 2024-11-12 at 10 23 56

**Review by theusaf — APPROVED** [review]

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Comment by luo3house** [comment]
draw glitches occurs while performing animated actions cuz blobs of animation frame inquired were quickly got gc
Screenshot 2024-11-14 at 14 03 21

**Review by Bas950 — APPROVED** [review]

**Pull request merged**